src/app/SafeAttributePersistenceProvider.h

```diff
@@ -50,7 +50,7 @@ class SafeAttributePersistenceProvider
      * @param [in] aValue the data to write.
      */
     template <typename T, std::enable_if_t<std::is_integral<T>::value, bool> = true>
-    CHIP_ERROR WriteScalarValue(const ConcreteAttributePath & aPath, T & aValue)
+    CHIP_ERROR WriteScalarValue(const ConcreteAttributePath & aPath, const T & aValue)
```

Comment:
Is there a reason this is pass-by-reference at all, not pass-by-value?

Reply:
Not that I can see. Changed it to by value.